src/database/to-dos.js

```diff
@@ -1,42 +1,55 @@
-const {getDatabase} = require('./mongo');
-const {ObjectID} = require('mongodb');
+const { getDatabase } = require('./mongo');
+const { ObjectID } = require('mongodb');
+const { paginate } = require('./../helper');
 
 const collectionName = 'to-dos';
 
 async function insertToDo(toDo) {
   const database = await getDatabase();
-  const {insertedId} = await database.collection(collectionName).insertOne(toDo);
+  const { insertedId } = await database
+    .collection(collectionName)
+    .insertOne(toDo);
   return insertedId;
 }
 
-async function getToDos() {
+async function getToDos(req) {
   const database = await getDatabase();
-  return await database.collection(collectionName).find({}).toArray();
+  const perPage = parseInt(req.query.per_page ? req.query.per_page : 20);
+  const rawData = database.collection(collectionName);
+  const currentPage = parseInt(req.query.page ? req.query.page : 1);
+  const total = await rawData.countDocuments();
+  const meta = paginate(perPage, currentPage, total);
+  const data = await rawData
+    .find({})
+    .skip(meta.nextSet)
+    .limit(perPage)
+    .toArray();
```

Comment:
I'm wondering if you _really_ need all this magic. For example, what would happen if instead of using this `paginate` function of your (which have a bad name since it doesn't paginate anything), and instead of using this `meta` object, you simply used `skip(currentPage * perPage)`?